common/src/main/java/io/seata/common/util/ReflectionUtil.java

```diff
@@ -243,4 +154,526 @@ public static void modifyStaticFinalField(Class<?> cla, String modifyFieldName,
+        if (staticField == null) {
+            throw new IllegalArgumentException("staticField must be not null");
+        }
+
+        // check is static field
+        if (!Modifier.isStatic(staticField.getModifiers())) {
+            throw new IllegalArgumentException("the `" + fieldToString(staticField) + "` is not a static field, cannot modify value.");
+        }
+
+        // remove the `final` keyword from the field
+        if (Modifier.isFinal(staticField.getModifiers())) {
+            Field modifiers = staticField.getClass().getDeclaredField("modifiers");
+            modifiers.setAccessible(true);
+            modifiers.setInt(staticField, staticField.getModifiers() & ~Modifier.FINAL);
+        }
+
+        // set new value
+        staticField.setAccessible(true);
+        staticField.set(staticField.getDeclaringClass(), newValue);
+    }
+
+    /**
+     * modify `static` or `static final` field value
+     *
+     * @param targetClass     the target class
+     * @param staticFieldName the static field name
+     * @param newValue        the new value
+     * @throws IllegalArgumentException if {@code targetClass} is {@code null}
+     * @throws NullPointerException     if {@code staticFieldName} is {@code null}
+     * @throws NoSuchFieldException     if the field named {@code modifyFieldName} does not exist
+     * @throws IllegalAccessException   the illegal access exception
+     */
+    public static void modifyStaticFinalField(Class<?> targetClass, String staticFieldName, Object newValue)
+            throws NoSuchFieldException, IllegalAccessException {
+        if (targetClass == null) {
+            throw new IllegalArgumentException("targetClass must be not null");
+        }
+
+        // get field
+        Field field = targetClass.getDeclaredField(staticFieldName);
+
+        // modify static final field value
+        modifyStaticFinalField(field, newValue);
+    }
+
+    //endregion
+
+
+    //region Method
+
+    /**
+     * get method
+     *
+     * @param clazz          the class
+     * @param methodName     the method name
+     * @param parameterTypes the parameter types
+     * @return the method
+     * @throws IllegalArgumentException if {@code clazz} is {@code null}
+     * @throws NullPointerException     if {@code methodName} is {@code null}
+     * @throws NoSuchMethodException    if the method named {@code methodName} does not exist
+     * @throws SecurityException        the security exception
+     */
+    public static Method getMethod(final Class<?> clazz, final String methodName, final Class<?>... parameterTypes)
+            throws NoSuchMethodException, SecurityException {
+        if (clazz == null) {
+            throw new IllegalArgumentException("clazz must be not null");
+        }
+
+        Class<?> cl = clazz;
+        while (cl != null) {
+            try {
+                return cl.getDeclaredMethod(methodName, parameterTypes);
```

Comment:
why not directly use getMethod?

Reply:
Because the `getMethod` can only get the `public` methods.